vertx-db2-client/src/main/java/io/vertx/db2client/impl/codec/DB2ParamDesc.java

```diff
@@ -40,12 +43,29 @@ public String prepare(TupleInternal values) {
         for (int i = 0; i < paramDefinitions.columns_; i++) {
         	Object val = values.getValue(i);
         	int type =  paramDefinitions.types_[i];
-        	if (!ClientTypes.canConvert(val, type)) {
+        	if (!canConvert(val, type)) {
         		return "Parameter at position [" + i + "] with class = [" + val.getClass() + 
         				"] cannot be coerced to [" + ClientTypes.getTypeString(type) + "] for encoding";
         	}
         }
         return null;
     }
+    
+    private static boolean canConvert(Object val, int type) {
+    	if (ClientTypes.canConvert(val, type))
+    		return true;
+    	Class<?> clazz = val.getClass();
+    	switch (type) {
+        case ClientTypes.BIGINT:
+        case ClientTypes.BOOLEAN:
+        case ClientTypes.DECIMAL:
+        case ClientTypes.DOUBLE:
+        case ClientTypes.INTEGER:
+        case ClientTypes.REAL:
+        case ClientTypes.SMALLINT:
+        	return clazz == Numeric.class;
+    	}
+    	return false;
+    }
 
```

Comment:
Why not just add "return clazz == Numeric.class" to canConvert in ClientTypes?  Is there a case where ClientTypes is a BIT, CHAR or LONGVARCHAR and you want to return a Numeric class only if it is not one of those types?

Reply:
right, the other types like CHAR, LONGVARCHAR, (and in the future BLOB and CLOB) won't be convertable to Numeric